openTSNE/nearest_neighbors.py

```diff
@@ -146,6 +146,92 @@ def query(self, query, k):
         return indices, distances
 
 
+class Annoy(KNNIndex):
+    VALID_METRICS = [
+        "cosine",
+        "euclidean",
+        "manhattan",
+        "hamming",
+        "dot",
+        "l1",
+        "l2",
+        "taxicab",
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__data = None
+        annoy_aliases = {
+            "cosine": "angular",
+            "l1": "manhattan",
+            "l2": "euclidean",
+            "taxicab": "manhattan",
+        }
+        if self.metric in annoy_aliases:
+            self.metric = annoy_aliases[self.metric]
```

Comment:
Please don't change `self.metric` here, but change it to the correct string at the last minute in `build`. The reason for this is that this won't work if the user manually changes the metric after initialization. Also, if they want to check which metric was used, they may be in for some confusion: 

```python
index = Annoy(metric="cosine")
print(index.metric)  # will be angular, not cosine
```

Reply:
ok done